### PR TITLE
Adds gun wield delay properly , makes it based off vigilance

### PR DESCRIPTION
**Pull request by MLGTASTICa**
## About The Pull Request
https://github.com/discordia-space/CEV-Eris/pull/6592
Redoes the  ^ PR properly , with the core principle of all mechanics related to combat being affected by stats
Maximum wield delay a gun can have is 2 SECONDS , this is substracted by the wield_delay / stat_vig * wield_factor,  wield factor is a variable that controls how much vigilance you need overall
Most shotguns need between 30-20 vigilance for instant wield , and take under 1 second to wield in general
Most rifles need 10-40 vigilance for instant wield , and take ~1 second to wield
Carbines/SMG need in general 20 vigilance and 0.2 seconds to wield , only exception is zoric with 0.3 seconds and 30 vigilance
Maxim/LMG/RPG have  2 second delay with 60 vigilance requirement for instant
Pistols have in general 0.1 second delay with 10 vigilance required , but people don't wield pistols
Revovlers have in general 0.2 seocnd delay and 20-30 vigilance required , same story as pistols however , they dont' get wielded

! SOME VALUES MIGHT STILL BE OFF , PLEASE LET ME KNOW IF YOU THINK SOME SHOULD BE TWEAKED, I HAD TO CHANGE THEM 2-3 TIMES TO GET IT PROPER !

L6 Saw and the Dallas also get more damage multiplier / pen multiplier , due to their rarity and wield_delays (L6 has 30% damage multiplier , 0.7 less recoil, Dallas has 20% more armor pen)

The wield delay reduction scales from 0 to the guns wield_delay_factor (0-20,40,60 vig depending on the gun)
## Why It's Good For The Game
Makes skills matter more overall in a way and adresses a few balance issues , predominantly instant gun pulling out , lack of reload delay , etc. Allows more avenues to balance guns around vigilance now, allows for more tactical combat
Ambushing can no longer be done at point blank properly / with being seen anymore

## Changelog
:cl:
balance : Guns in general now have a wield delay that is based off their firepower and mag size , wield delay is at maximum 2 SECONDS and that is reached by LMGS / RPG , and it gets reduced by the users vigilance stat
/:cl:

### CODE REVIEW

**Comment by Kegdo** [comment]
Makes ambushes even more dangerous, and makes pistols even better(they aren't meta tho), and while I use only pistols, I really hate being ambushed, so that's a negative ghostrider

**Comment by MLGTASTICa** [comment]
> Makes ambushes even more dangerous, and makes pistols even better(they aren't meta tho), and while I use only pistols, I really hate being ambushed, so that's a negative ghostrider //////// ALSO PLEASE, **PLEASE** LET'S NOT MAKE THIS TGMC, WIELD DELAY SUCKS AND ALWAYS WILL SUCK, THIS IS A BAD IDEA

Ok (pulls LMG at PB instantly and blasts you)

**Comment by Kegdo** [comment]
> > Makes ambushes even more dangerous, and makes pistols even better(they aren't meta tho), and while I use only pistols, I really hate being ambushed, so that's a negative ghostrider //////// ALSO PLEASE, **PLEASE** LET'S NOT MAKE THIS TGMC, WIELD DELAY SUCKS AND ALWAYS WILL SUCK, THIS IS A BAD IDEA
> 
> Ok (pulls LMG at PB instantly and blasts you)

As long as it's not an ERP filled /me verb, I will die proudly

**Comment by Humonitarian** [comment]
I like the idea of wield delay, but not the values.

I'd rather keep vigilance limited with diminishing returns, but all delays very short, with not even RPGs more than a second.

Instant-wield should be limited to holsters, for the western gags.

Don't forget by the one-hour mark majority of people have at least 30-40 vigilance as well, and most IH start with 20-30 roundstart.

**Comment by MLGTASTICa** [comment]
> I like the idea of wield delay, but not the values.
> 
> I'd rather keep vigilance limited with diminishing returns, but all delays very short, with not even RPGs more than a second.
> 
> Instant-wield should be limited to holsters, for the western gags.
> 
> Don't forget by the one-hour mark majority of people have at least 30-40 vigilance as well, and most IH start with 20-30 roundstart.

Diminishing returns doesn't fit here at all. This is gun wielding. Diminishing return is fitting for things that have no limits like punch damage , etc , but here the limits are clearly made to be surpassed as a reward for having the skill.

**Comment by Humonitarian** [comment]
> Diminishing returns doesn't fit here at all. This is gun wielding. Diminishing return is fitting for things that have no limits like punch damage , etc , but here the limits are clearly made to be surpassed as a reward for having the skill.
Then add a minimum. Setting the RPG at 60 doesn't do anything, serbs already start at around that much vigilance, allowing them to effectively instant-wield it from get-go. This goes for many other weapons in many other scenarios.

> Also RPGs should have a lengthy delay , they are a instant win if they score a hit off (not really great fun for the otherside)

Tell that to Presidente. (He once soloed two serbs after getting hit by an RPG point-blank)

**Comment by MLGTASTICa** [comment]
> > Diminishing returns doesn't fit here at all. This is gun wielding. Diminishing return is fitting for things that have no limits like punch damage , etc , but here the limits are clearly made to be surpassed as a reward for having the skill.
> 
> Then add a minimum. Setting the RPG at 60 doesn't do anything, serbs already start at around that much vigilance, allowing them to effectively instant-wield it from get-go. This goes for many other weapons in many other scenarios.
> 
> > Also RPGs should have a lengthy delay , they are a instant win if they score a hit off (not really great fun for the otherside)
> 
> Tell that to Presidente. (He once soloed two serbs after getting hit by an RPG point-blank)

serbs start with 20-30 vigilance depending on background
As for presidente , luck.

**Comment by Humonitarian** [comment]
> serbs start with 20-30 vigilance depending on background

Job gives 30, actually.
That's a variation between 20-40, usually the higher end knowing the average background user

**Comment by drexample** [comment]
Skimmed the code, the effective in-game numbers seem fine, ready for more thorough review

**Comment by ValoTheValo** [comment]
> > Makes ambushes even more dangerous, and makes pistols even better(they aren't meta tho), and while I use only pistols, I really hate being ambushed, so that's a negative ghostrider //////// ALSO PLEASE, **PLEASE** LET'S NOT MAKE THIS TGMC, WIELD DELAY SUCKS AND ALWAYS WILL SUCK, THIS IS A BAD IDEA
> 
> Ok (pulls LMG at PB instantly and blasts you)

skill issue.

**Comment by FortuneMaster** [comment]
:scream cat: dont make it based of skill, people with lower skill and ambushed are already at a disadvantage, no point in making the gap even bigger

File: code/modules/projectiles/guns/projectile/battle_rifle/type47.dm [hunk 1]
```diff
@@ -17,14 +17,15 @@
...
 	damage_multiplier = 1.5
-	one_hand_penalty = 10 
+	one_hand_penalty = 10
 	gun_tags = list(GUN_SILENCABLE)
 	gun_parts = list(/obj/item/part/gun = 2 ,/obj/item/stack/material/plasteel = 6)
 	init_firemodes = list(
 		SEMI_AUTO_NODELAY,
 		BURST_3_ROUND
 		)
-	spawn_blacklisted = TRUE //until loot rework
```
Comment (hyperioo) [inline code/modules/projectiles/guns/projectile/battle_rifle/type47.dm:27]: Sneaky unblacklist of the type47?

File: code/modules/projectiles/guns/projectile/battle_rifle/boltgun.dm [hunk 2]
```diff
@@ -32,6 +32,8 @@
 	sawn = /obj/item/gun/projectile/boltgun/obrez
 	var/bolt_open = 0
 	var/item_suffix = ""
+	wield_delay = 0.3 SECOND
+	wield_delay_factor = 0.2 // 40 vig
```
Comment (hyperioo) [inline code/modules/projectiles/guns/projectile/battle_rifle/boltgun.dm:36]: 0.4 for 40 vig or put `// 20 vig`

File: code/modules/projectiles/guns/projectile/automatic/vintorez.dm [hunk 3]
```diff
@@ -25,6 +25,8 @@
 		SEMI_AUTO_NODELAY,
 		FULL_AUTO_400
 		)
+	wield_delay = 1.5 SECOND
+	wield_delay_factor = 0.4 // 30 vig to insta wield , heavy class rifle
```
Comment (hyperioo) [inline code/modules/projectiles/guns/projectile/automatic/vintorez.dm:29]: 0.4 != "30 vig"

File: code/modules/projectiles/guns/projectile/automatic/drozd.dm [hunk 4]
```diff
@@ -24,6 +24,9 @@
 	one_hand_penalty = 5 //smg level
 	gun_tags = list(GUN_SILENCABLE)
 
+	wield_delay = 0.5 SECOND
+	wield_delay_factor = 0.1 // 20 vig
```
Comment (hyperioo) [inline code/modules/projectiles/guns/projectile/automatic/drozd.dm:28]: 0.1 != 20 vig

File: code/modules/projectiles/guns/projectile/automatic/c20r.dm [hunk 5]
```diff
@@ -27,6 +27,8 @@
 	zoom_factor = 0.4
 	recoil_buildup = 1.2
 	one_hand_penalty = 5 //smg level
+	wield_delay = 0.5 SECOND
+	wield_delay_factor = 0.1 // 20 vig
```
Comment (hyperioo) [inline code/modules/projectiles/guns/projectile/automatic/c20r.dm:31]: 0.1 != 20 vig

File: code/modules/projectiles/guns/energy/nuclear.dm [hunk 6]
```diff
@@ -21,6 +21,8 @@
 		)
 
 	var/lightfail = 0
+	wield_delay = 1 SECOND
+	wield_delay_factor = 0.3 // 20 vig
```
Comment (hyperioo) [inline code/modules/projectiles/guns/energy/nuclear.dm:25]: 0.3 != 20 vig

File: code/modules/projectiles/guns/energy/lasersmg.dm [hunk 7]
```diff
@@ -29,6 +29,9 @@
 		FULL_AUTO_300
 		)
 
+	wield_delay = 1 SECOND
+	wield_delay_factor = 0.1 // 20 vig
```
Comment (hyperioo) [inline code/modules/projectiles/guns/energy/lasersmg.dm:33]: 0.1 != 20 vig

File: code/modules/projectiles/gun.dm [hunk 8]
```diff
@@ -89,6 +89,21 @@
...
+	if(!wield_delay)
+		..()
+		return
+	var/calculated_delay = wield_delay
+	if(ishuman(user))
+		calculated_delay = wield_delay - (wield_delay * (user.stats.getStat(STAT_VIG) /( 100 * wield_delay_factor))) // wield delay - wield_delay * user vigilance / 100 * wield_factor
+	if (calculated_delay > 0 && do_after(user, calculated_delay, mobile = TRUE))
+		..()
+	else if (calculated_delay < 0)
+		..()
```
Comment (hyperioo) [inline code/modules/projectiles/gun.dm:105]:
```suggestion
	if(ishuman(user))
		calculated_delay = wield_delay - (wield_delay * (user.stats.getStat(STAT_VIG) / (100 * wield_delay_factor))) // wield delay - wield_delay * user vigilance / 100 * wield_factor
	if (calculated_delay > 0 && do_after(user, calculated_delay, mobile = TRUE))
		..()
	else if (calculated_delay <= 0)
		..()
```

**Review by hyperioo — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/projectile/battle_rifle/type47.dm (hunk 1 above)
Comment (MLGTASTICa) [reply code/modules/projectiles/guns/projectile/battle_rifle/type47.dm:27]: i love fixing merge conflicts i love fixing merge conflicts i love fixing merge conflicts

**Review by drexample — APPROVED** [review]
Let's see how many things this will break.
Gun related code seems to love doing that.

**Pull request merged**